### PR TITLE
HTML: Don't preserve line breaks around text-only content

### LINKED ISSUE #8287: Inconsistent formatting on Vue double curly braces template variables
**Prettier 2.0.5**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeeBbADgGwIbwB8AOlAARmoAmAlgG5lj4DOzAvMSAE5xWcnkKZAOpwyeHmQDmEGlClkYEcVSqKAFjWZlmNbDjFgJa9XB6kKqAPS06Ay7cYt2nHnxD2hw9QTJbpsvJk0GTAwGRYXBBYAIxkAL7x4lBqYRFRWABMCfEWlDb0AtaYuARwAiAANCDRMDTQzMigElEA7gAKEgiNKHg4rXgAno3VAEZceGAA1nAwAMpYk3JSyDBcAK5w1eowGDjeNPDMi2Bwc92H9IeDyCA0UlAQPFV3UMxmMO0TUhh4yABmfXe1QAVswAB4AIQm01mczwGDgABk5HAAUCtiAweC5ssDABFdYQeDonDAkCLLjvLi3OibF6ROQwYQ0KgwdTIAAcAAZqpEIO9hBMsLdInBqXQ0dUAI5E+BfaI9EB4ZgAWigcF4vBePFlNB4XzwPz+SEBZMx7wwNFWGwteLghOJaNNGOqMDwoxZbI5SEybomNBwywAwhAML9buKAKwvdbvAAqHp6ZvJdLgAEkUgh5mAuDQsDAAIIpOYwQYGUnvRJAA)
```sh
--html-whitespace-sensitivity ignore
--parser vue
```

**Input:**
```vue
<template>
  <div class="red">
    We are going to add this simple card here
  </div>
  <div class="red">
    What is going on {{ prop1 }} and {{ prop2 }}
  </div>
</template>

```

**Output:**
```vue
<template>
  <div class="red">
    We are going to add this simple card here
  </div>
  <div class="red">What is going on {{ prop1 }} and {{ prop2 }}</div>
</template>

```

**Expected behavior:**
I expected that it makes no difference between the first and second div block in terms of formatting.  Only when I add prop values do i get the error/recommendation to keep everything on one line.  If I switch the parser to HTML it will not attempt to change or throw the error.  This is when working within .vue component files.

### PR DESCRIPTION
**Pull request by fisker**
Fixes #8287

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [ ] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/language-html/utils.js [hunk 1]
```diff
@@ -287,7 +287,6 @@ function forceBreakContent(node) {
       (["body", "script", "style"].includes(node.name) ||
         node.children.some((child) => hasNonTextChild(child)))) ||
     (node.firstChild &&
-      node.firstChild === node.lastChild &&
```
Comment (thorn0) [inline src/language-html/utils.js:290]: Any idea on why we had this condition?

Comment (fisker) [reply src/language-html/utils.js:290]: Don't know, but I think this mean to test text node is the only node?

Comment (fisker) [reply src/language-html/utils.js:290]: These should has no difference.

**Prettier 2.0.5**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAdKACDAeAJgSwDcMwAbAQwGdKBeVEAJzl3oD50ssB1ODcpjAHMI+KIIwwIfXLgkALfJQyV8AWwAOpXmH6y5cJhxwB6AoXaYcZkhWp1GzNke5zyMDIqEixGaDkrq5FCs6gwQ6gCM2MYBQRZY0WYWIAA0IOEw+NCUyKD8YQDuAAr8CDko5KQF5ACeOWkARgzkYADWcDAAyoFgooLIMAwArnBpcjCqpFwK8LFgcJ1l+JmEyzXI4NSpIKKUBjBFzYKq5MgAZpV7aQBWlAAeAELNbR2d5KpwADKicOeXoyBbndOn0tABFIYQeB-UhXECBBh7BgbcaTbahUQwLj4XAwOTIAAcAAY0qEIHsuM11BtQnAkYRfmkAI6Q+CHcLlEBUAC0UDgzGY2yYLPwTEO5GOpyQF1hAL2qnwA2GctBcAhUN+0v+aRg5Aa2Nx+KQACYdc18KQ+gBhCCqE4bOkAVm2Qz2ABU9eUZXDCCMAJJQXAILpgBj4dQwACCgc6MBqWhhewAvkmgA)
```sh
--parser html
```

**Input:**
```html

  <div class="red">
    We are going to add this simple card here
  </div>
  <div class="red">
    What is going on <span>prop1</span>
  </div>

```

**Output:**
```html
<div class="red">
  We are going to add this simple card here
</div>
<div class="red">What is going on <span>prop1</span></div>

```

**Prettier pr-8614**
[Playground link](https://deploy-preview-8614--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEAdKACDAeAJgSwDcMwAbAQwGdKBeVEAJzl3oD50ssB1ODcpjAHMI+KIIwwIfXLgkALfJQyV8AWwAOpXmH6y5cJhxwB6AoXaYcZkhWp1GzNke5zyMDIqEixGaDkrq5FCs6gwQ6gCM2MYBQRZY0WYWIAA0IOEw+NCUyKD8YQDuAAr8CDko5KQF5ACeOWkARgzkYADWcDAAyoFgooLIMAwArnBpcKoNzLjMADJBgkPkgnAAYhAMqm6ZYsgg5EOSqSByMKqkXArwsWBwnWX4mYQPNbtg1EeilAYwRc2Cm8gAGaVL5pABWlAAHgAhZptDqdciqOAzURwIEg0YgCGQzp9LQARSGEHgGNIoJAgQYXwYuxOZyOoVEMC4+FwMDkyAAHAAGNKhCBfLjNdS7UJwGmEdFpACOxPgv3C5T2lAAtFA4MxmEcmHL8ExfksAUhgeSsV9VPgBsNzfi4ESSeiTZi0jByA1WezOUgAEyu5r4Uh9ADCEFUxpAEoArEchl8ACru8qmimEEYASSg01gnTADHw6hgAEEs50YDUtGSvgBfatAA)
```sh
--parser html
```

**Input:**
```html

  <div class="red">
    We are going to add this simple card here
  </div>
  <div class="red">
    What is going on <span>prop1</span>
  </div>

```

**Output:**
```html
<div class="red">
  We are going to add this simple card here
</div>
<div class="red">
  What is going on <span>prop1</span>
</div>

```

Comment (fisker) [reply src/language-html/utils.js:290]: https://github.com/prettier/prettier/pull/5596/files

Comment (thorn0) [reply src/language-html/utils.js:290]: #5596 fixed issues with line breaks around elements. Do we really need to preserve line breaks around text?

Are we sure the following output wouldn't be better in #8287?
```html
<template>
  <div class="red">We are going to add this simple card here</div>
  <div class="red">What is going on {{ prop1 }} and {{ prop2 }}</div>
</template>
```

Comment (fisker) [reply src/language-html/utils.js:290]: Seems we are checking leading line breaks

**Prettier 2.0.5**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAdKACDAeAJgSwDcMwAbAQwGdKBeVEAJzl3oD4B1ODcpjAcwj4ofDDAjdcuUQAt8lDJXwBbAA6kuYHlOlwm6dNgD0BQq3RY8REhWp1GzNu2nkYGOf0HCM0HJRXkoVhUGCBUARiM-ALMoA2MiGMtiMipaeiYWEA5nV3cBIREfYGAMYNCwjABfSu4oKWLSkJUAJirK8ww4kxiQABoQUJh8aEpkUB4QgHcABR4EUZRyUknyAE9R-oAjBnIwAGs4GABlfzAC5BgGAFc4fukYJVInfHgosDgj+ZeiF9XkcGofRAQkouhg0x2fCU5GQADMlqD+gArSgADwAQjt9ocjuQlHAADJCOBwhG3EAo1FHArqACKVwg8FJpERIH8DFBDH+90eQOCQhg7HwuBg0mQAA4AAz9Mqg9g7FT-YJwTmEEn9ACODPgENCCxAVAAtFA4MxmECmFr8EwIeQoTCkPCWeTQUp8Bdri6aXB6YySY6yf0YORNkKRWKkM0gzt8KQCgBhCBKaH-FUAViBV1BABUQwsnazCDcAJJ1BDHMAMfAqGAAQTqRxgq3UzNB1SAA)
```sh
--parser html
```

**Input:**
```html

  <div class="red">We are going to add this simple card here

</div>
  <div class="red">What is going on <span>prop1</span>

</div>
<div class="red">What is going on {{ prop1 }} and {{ prop2 }}
  
</div>

```

**Output:**
```html
<div class="red">We are going to add this simple card here</div>
<div class="red">What is going on <span>prop1</span></div>
<div class="red">What is going on {{ prop1 }} and {{ prop2 }}</div>

```

Comment (thorn0) [reply src/language-html/utils.js:290]: Yes, see https://github.com/prettier/prettier/pull/5596#issue-236042241

Comment (thorn0) [reply src/language-html/utils.js:290]: The question is: do we want to do that for text (with or without interpolations)? It was done to solve issues with elements, not with text.

Comment (fisker) [reply src/language-html/utils.js:290]: What about this `<div class="red">What is going on <span>prop1</span></div>`. It's element.

Comment (fisker) [reply src/language-html/utils.js:290]: Could be a icon `<div class="red">What is going on <i class="icon"></i></div>`

Comment (thorn0) [reply src/language-html/utils.js:290]: I mean the cases where the element contains only text (probably with interpolations), like in #8287. If there are elements, we can keep applying the rules from https://github.com/prettier/prettier/pull/5596#issue-236042241. But if there is only text?

Comment (fisker) [reply src/language-html/utils.js:290]: I'll check.

Comment (fisker) [reply src/language-html/utils.js:290]: I've updated, but feel less readable for long text.

File: changelog_unreleased/html/pr-8614.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+</div>
+
+<!-- Prettier stable -->
+<div>
+  Hello, world!
+</div>
+
+<!-- Prettier master -->
+<div>Hello, world!</div>
+```
```
Comment (evilebottnawi) [inline changelog_unreleased/html/pr-8614.md:17]: Maybe we can improve it, it’s not clear what really changed and why?

Comment (thorn0) [reply changelog_unreleased/html/pr-8614.md:17]: Yes, this needs an explanation. I can try to write it when I have time this weekend.

Comment (fisker) [reply changelog_unreleased/html/pr-8614.md:17]: Thank you.

**Comment by fisker** [comment]
Thank you for the changelog.

**Pull request merged**